app/utils.py

```diff
@@ -152,25 +152,44 @@ def setup_logger(name, level=logging.INFO):
     return logger
 
 
-def validate_resource(json):
+def validate_resource(request, id=-1):
+    json = request.get_json()
     validation_errors = {'errors': {'type': 'validation'}}
+    missing_params = []
+    invalid_params = []
+
     if not json:
         message = "A JSON body is required to use this endpoint, but none was given"
         validation_errors['errors']['message'] = message
         return validation_errors
 
-    validation_errors['errors']['missing_params'] = []
-
-    required = []
-    for column in Resource.__table__.columns:
-        if column.nullable is False and column.name != 'id':
-            # strip _id from category_id
-            name = column.name.replace('_id', '')
-            required.append(name)
-
-    for prop in required:
-        if json.get(prop) is None:
-            validation_errors['errors']['missing_params'].append(prop)
-
-    if validation_errors['errors']['missing_params']:
+    # There are only required parameters for creating new Resources
+    if request.method == 'POST':
+        required = []
+        for column in Resource.__table__.columns:
+            if column.nullable is False and column.name != 'id':
+                # strip _id from category_id
+                name = column.name.replace('_id', '')
+                required.append(name)
+
+        for prop in required:
+            if json.get(prop) is None:
+                missing_params.append(prop)
+
+    url = json.get("url")
+    if url is not None:
+        # If there is an existing resource with the requested url and it isn't the Resource we are trying to update.
+        resource = Resource.query.filter_by(url=url).first()
+        if resource and resource.id != id:
+            invalid_params.append("url")
+
+    if missing_params:
+        validation_errors['errors']['missing_params'] = []
```

Comment:
I don't think this line is necessary. You're setting `missing_params = []` up on line 158, so the next line will have the same effect if there are no missing params. Same with line 191, seems unnecessary.